server/src/main/java/io/crate/execution/dml/Indexer.java

```diff
@@ -348,11 +365,13 @@ public Indexer(String indexName,
                    NodeContext nodeCtx,
                    Function<ColumnIdent, FieldType> getFieldType,
                    List<Reference> targetColumns,
-                   Symbol[] returnValues) {
+                   Symbol[] returnValues,
+                   Version minNodeVersion) {
         this.symbolEval = new SymbolEvaluator(txnCtx, nodeCtx, SubQueryResults.EMPTY);
         this.columns = targetColumns;
         this.synthetics = new HashMap<>();
         this.stream = new BytesStreamOutput();
+        this.targetsHaveOids = table.versionCreated().onOrAfter(Version.V_5_5_0) && minNodeVersion.onOrAfter(Version.V_5_5_0);
```

Comment:
I don't think it's safe to use this condition to decide if oid indexing should be used.

A mixed cluster is temporary, so you can end up with a table that has some documents indexed with oids and some without.

I think to avoid the streaming dependency we could look up the oids in the local state? The cluster state should have updated locally too if we got a response from the add-column action.

Reply:
As discussed internally in https://crate.slack.com/archives/CBPLN6WJK/p1693396898974569,
we can only rely on Version.created.

Done in https://github.com/crate/crate/pull/14617/commits/b45283b1878eed710f265d903389204579303163. 

Also, in a follow up PR, where I will be actually using OID, we will have kind of `Function<Reference, String> sourceKeyWriter,` which would be either `ref -> ref.column.leafName` or `Long.toString(ref.oid())`, depending on version.created

cc @seut